### DISCUSSION TITLE
Getting panic: +2/3 committed an invalid block: wrong Block.Header.AppHash

### DISCUSSION THREAD

**Discussion by aasierra** [crypto-org-chain/cronos#299] [Q&A]
Hi there, first thank you for taking a look at this. I have a question about whether what I am doing should work or not.

I cloned the repository, then ran `git checkout v0.6.5`. I then ran the `make build NETWORK=testnet`

Next I ran `./build/cronosd start`

I made sure that I downloaded the correct testnet genesis file but I am coming across this issue.

panic: +2/3 committed an invalid block: wrong Block.Header.AppHash.  Expected 702DC60359362E3D03038E2C58EB6BC176574474848B3C7C2F93EE0B5B45069B, got 61BAA08C8C960B8873DAD2760E38DE73AC6DBBFC46A840DB435F20616AB570B5

I have tried various fixes as well as going through the various issues and trying some of the fixes there. I was posting here to hopefully get some help with this. I can go ahead and post my data and config files if that helps.

[genesis.json](https://gist.github.com/aasierra/689572e0a0dad1c7f210639577e7a5db)
[config.toml](https://gist.github.com/aasierra/bea66e8b2acb08e7411272bd76eb24d1)

Thank you,
Anthony

**Comment by yihuang** [comment]
I think you need this version of binary to run testnet node: https://github.com/crypto-org-chain/cronos/releases/tag/v0.6.0-testnet

**Reply by aasierra** [reply]
Hello @yihuang I appreciate the response. After looking more into this, I was able to run on the mainnet. I have not had time to test with the test net again. 

I apologize, I should have stated this earlier. The issue I am running into is me attempting to run and build the code as if I was developing for it. Since it is working on mainnet, I assume there was a configuration I did not set correctly in the make file. I have not had time to investigate it more and if I have time I will attempt to post here a solution.

**Reply by aasierra** [reply]
Just updating here, it happened on the mainnet as well. I think am going to have to redownload everything again as my WAL has no indication of  CORRUPTED MESSAGE to repair it.
